index.js

```diff
@@ -38,7 +38,7 @@ checker.init({
       });
 
       // Generate report with packages containing the licenses passed to `failOn`
-      if (!argv.disableErrorReport) {
+      if (!argv.disableReport && !argv.disableErrorReport) {
```

Comment:
@LonelyPrincess , for me it's a little bit confusing having to use a double flag to disable generating the error report.

As we already have `disableErrorReport` flag, why should we need to set this new flag to avoid generating the error report?

Reply:
`disableErrorReport` only prevents error reports from being created, but it doesn't avoid the report generation when no error ocurrs. This new flag, however, aims to avoid generating reports in general, whether the operation ends up in success or an error. This is the desired behaviour that was described in issue #12. 

This condition here just checks if any of those flags are currently activated, as both of them should stop the report from being generated.

Reply:
I understand your point. However, the issue #12 was added by someone requesting that feature, but has not been discussed internally. In any case, if we want to add that functionality, the condition them should be an OR instead of an AND:

`if (!argv.disableReport || !argv.disableErrorReport)`

Otherwise, we would be forcing the user to manually set both flags to true.

In my opinion, the feature requested in issue #12 has conflicts with the current behaviour of the CLI commands. We could add the flag `--disableReport` or perhaps `--disableOutput` to avoid generating the license file even if the check is OK. But we already have a flag to avoid the report when there is an error. So if a user wants to avoid having any kind of report, it doesn't matter the result, they would use something similar to:

`--disableErrorReport --disableOutput``

What do you think @LonelyPrincess ?

Reply:
Thanks for the feedback, Iñigo!

> The issue #12 was added by someone requesting that feature, but has not been discussed internally.

If you guys don't consider it useful, you can tag that issue as "won't fix", close it and also close this PR to avoid further confusion. 👍🏻 

> If we want to add that functionality, the condition them should be an OR instead of an AND`. Otherwise, we would be forcing the user to manually set both flags to true.

No, that's not really what's happening here. If you look at the condition more closely, what we're checking here is that both of the flags are `false`.

```
if (!argv.disableReport && !argv.disableErrorReport) { /* generate report */ }
```

So you can see it clearer, we could translate this expression to this: "if both flags are disabled, generate report" / "if none of the flags is enabled, generate report". 

If we changed it to this to what you propose...

```
if (!argv.disableReport || !argv.disableErrorReport) { /* generate report */ }
```

... what would happen is that the report would be generated if any of the two flags is set to `false`. Basically, your expression can be read as: "if at least one of the flags is disabled, generate report". The results for this condition are listed below:

| `disableReport` | `disableErrorReport` | Is report generated? |
| --- | --- | --- |
| `true` | `true` | `false` |
| `true` | `false` | `true` |
| `false` | `true` | `true` |
| `false` | `false` | `true` |

As you can see, the report would be generated in two scenarios where it should not be created. If `disableErrorReport` is enabled, we do not want the error report to be created regardless of the current value of the `disableReport` flag. Same logic applies the other way round. 

> So if a user wants to avoid having any kind of report, it doesn't matter the result, they would use something similar to:
> ```--disableErrorReport --disableOutput```

Hmmm... 🤔 I'm not sure about this, really. From what I understand, the user won't normally use the two flags together, as one is basically an extension of the other.

- `--disableReport` will be added when they want no output to be created.
- `--disableErrorReport` will be added when they want no output to be created only in case an error happens.

Potentially an user could add both as in your example, but it wouldn't make much sense as the `--disableReport` already implies that an error report won't be generated.